WORKSPACE

```diff
@@ -33,6 +33,32 @@ load("@os_info//:os_info.bzl", "is_linux", "is_windows")
 # bazel dependencies
 haskell_repositories()
 
+load("@io_tweag_rules_haskell//haskell:cabal.bzl", "stack_install")
+
+stack_install(
+    name = "stackage",
+    packages = [
+        # Core libraries
+        "array",
+        "base",
+        "directory",
+        "filepath",
+        "process",
+        # For tests
+        "streaming",
+        "void",
+        "hspec",
+        "hspec-core",
+        "lens-family-core",
+        "data-default-class",
+        "lens-labels",
+        "proto-lens",
+        "lens-family",
+    ],
+    snapshot = "lts-13.15",
+    deps = ["@zlib.dev//:zlib"],
```

Comment:
How do we know which package to forward these to? Or are they generally visible to all packages? In the latter case that means that adding a `deps` entry here would rebuild all imported stackage packages, right?

Reply:
I can answer this myself now. Yes, they are forward to all packages.

Reply:
We can,

* ask the user to specify exactly which package we should expose the system lib to,
* blanket forward the dependency to all packages.

The latter means more rebuilds as you say. I like minimal incremental rebuilds. But OTOH with the latter option we no longer have a conventional `deps` attribute with a conventional type (label list). Moreover, as Hazel taught me, Cabal seems to expose a lot of C stuff to all transitive downstream dependencies. So if we change the set of system libs, we change the inputs for most of the dependency graph and therefore have to rebuild most of it anyways. It's perhaps why in both Stack and Cabal, this is a global option that affects the inputs to all packages.